src/convertToObject.js

```diff
@@ -6,7 +6,24 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const KeyAndValues = sourceString.split(';');
+  const end = {};
+
+  KeyAndValues.forEach((element) => {
+    const preElement = element.split(':');
+
+    if (preElement[0]) {
+      preElement[0] = preElement[0].trim();
+    }
+
+    if (preElement[1]) {
+      preElement[1] = preElement[1].trim();
+    }
+
+    end[preElement[0]] = preElement[1];
```

Comment:
This line will add undefined as a value in the 'end' object if 'preElement[1]' is undefined. You should add a check to ensure both the key and value exist before adding them to the 'end' object.